lib/src/single_subscription_transformer.dart

```diff
@@ -30,6 +28,7 @@ class SingleSubscriptionTransformer<S, T> extends StreamTransformerBase<S, T> {
         controller.addError(error, stackTrace);
       }
     }, onError: controller.addError, onDone: controller.close);
+    controller.onCancel = subscription.cancel;
```

Comment:
You can avoid a lot of `late` by splitting creation and initialization when you have cyclic dependencies!